docs/book/advanced-guide/pipelines/gpu-hardware.md

```diff
@@ -0,0 +1,57 @@
+---
+description: How to ensure your pipelines or steps run on GPU-backed hardware
+---
+
+ZenML allows for multiple ways to configure the hardware on which your steps
+run, from [step operator stack
+components](../../component-gallery/step-operators/step-operators.md) to [custom
+per-step or per-pipeline
+requirements](../../advanced-guide/pipelines/settings.md). For steps or
+pipelines that are required to run on GPUs, it is essential to ensure that the
+environment has the required CUDA tools installed. The following section describes what
+you need to do to ensure that you will actually get the performance boost that
+running your training on a GPU will give you.
+
+The steps that will run on GPU-backed hardware will all be running from a
+containerized environment, whether you're using our local Docker orchestrator or
+on a cloud instance of Kubeflow. (Please see [the section on configuration of the
+Docker environment](../../advanced-guide/pipelines/containerization.md) for
+general context on this and what follows.) For this reason, you will need to
+make two amendments to your Docker settings for the relevant steps as follows:
+
+1. Specify a CUDA-enabled parent image in your `DockerSettings`
+
+For full details, see the whole section where we explain how to do this [on the
+containerization page](../../advanced-guide/pipelines/containerization.md). As
+an example, if you wanted to use the latest CUDA-enabled official PyTorch image
+for your entire pipeline run, you could include the following code:
+
+```python
+docker_settings = DockerSettings(parent_image="pytorch/pytorch:latest")
+
+@pipeline(settings={"docker": docker_settings})
+def my_pipeline(...):
+    ...
+```
+
+2. Add ZenML as an explicit pip requirement
```

Comment:
why is this necessary, don't we do this under the hood when building images? Or, asked differently, why don't we need to do this when running on CPU?

Reply:
Not entirely sure, but this is what @schustmi told me this morning. Maybe he has the answer. I think step #2 is just precautionary and perhaps not essential?

Reply:
You always need ZenML installed in every case, because the code that runs inside the docker image is an entrypoint file which is inside ZenML.That code actually kicks off the pipeline.

In the `CPU` case, you dont specify a parent_image @fa9r, so you the default zenml base image is used which has zenml already installed. With this current approach, a pytorch base image with cuda is being used and of course that means zenml needs to be installed. So its quite mandatory